jitterutils.c

```diff
@@ -31,6 +31,87 @@
 
 #include "jitterdebugger.h"
 
+
+struct ringbuffer_sample {
+	struct timespec ts;
+	uint64_t val;
+};
+
+struct ringbuffer {
+	uint32_t size;
+	uint32_t overflow;
+	uint32_t read;
+	uint32_t write;
+	struct ringbuffer_sample *data;
+};
+
+static int ringbuffer_full(uint32_t size, uint32_t read, uint32_t write)
+{
+	return write - read == size;
+}
+
+static int ringbuffer_empty(uint32_t read, uint32_t write)
+{
+	return write == read;
+}
+
+static uint32_t ringbuffer_mask(uint32_t size, uint32_t idx)
+{
+	return idx & (size - 1);
+}
```

Comment:
maybe inline those three guys?

Reply:
I rather leave them as documentation there. Sure it's 'easy' to understand the expression but I still prefer the readable string.